src/Resource.ts

```diff
@@ -49,20 +85,38 @@ export class Resource<T extends SimpleObject> {
     body?: T;
     query?: SimpleObject;
   }): Promise<any> {
-    const promise = this.axiosInstance({
+    const request = this.axiosInstance({
       method,
       url: `${this.name}/${path}`,
       data: body && convertObjectToSnakeCase(body),
       params: query && convertObjectToSnakeCase(query)
     });
 
-    try {
-      const response = await promise;
+    return handleResponse(request);
+  }
+
+  protected async upload(body: T): Promise<any> {
+    const formData = toFormData(body);
+
+    const request = this.axiosInstance({
+      method: Method.POST,
+      url: `${this.name}/`,
+      data: formData,
+      headers: formData.getHeaders()
+    });
+
+    return handleResponse(request);
+  }
+
+  protected async download(id: string): Promise<OnfidoDownload> {
+    const request = this.axiosInstance({
+      method: Method.GET,
+      url: `${this.name}/${id}/download`,
+      responseType: "stream",
+      headers: { Accept: "*/*" }
```

Comment:
In `Onfido.ts` I set the default `Accept` header to `application/json`, but for download it could be anything really (`image/png`, `application/pdf`, `video/mp4`, ...)

Reply:
Worth adding a little comment 😄